freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -550,12 +550,25 @@ void applyStateDifferenceToIpa(String environmentCrn, FreeIpaClient freeIpaClien
 
     void addGroups(boolean fmsToFreeipaBatchCallEnabled, FreeIpaClient freeIpaClient, Set<FmsGroup> fmsGroups,
             BiConsumer<String, String> warnings) throws FreeIpaClientException {
-        List<GroupAddOperation> operations = Lists.newArrayList();
+        List<GroupAddOperation> posixOperations = Lists.newArrayList();
+        List<GroupAddOperation> nonPosixOperations = Lists.newArrayList();
         for (FmsGroup fmsGroup : fmsGroups) {
-            operations.add(GroupAddOperation.create(fmsGroup.getName(), warnings));
+            String groupName = fmsGroup.getName();
+            // batch calls that contain both posix and non-posix groups only created the posix groups
+            if (isNonPosixGroup(groupName)) {
```

Comment:
why is this separation necessary?

Reply:
This is empirical information. I initially tried to include both posix and non-posix in a single batch request but the non-posix group was not created. Splitting the operations into two requests did work. I don't know if this is an issue with the batch api, but I didn't pursue it further.

Reply:
could you add this information to the commit message please? it would be very useful later. I prefer those instead of comments

Reply:
I think only this one is left, otherwise LGTM

Reply:
thanks! I removed the comment and updated the commit message